src/node/utxo_snapshot.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/utxo_snapshot.h>
+
+#include <fs.h>
+#include <logging.h>
+#include <streams.h>
+#include <uint256.h>
+#include <validation.h>
+
+#include <cstdio>
+#include <optional>
+
+namespace node {
+
+bool WriteSnapshotBaseBlockhash(CChainState& snapshot_chainstate)
+{
+    AssertLockHeld(::cs_main);
+    assert(snapshot_chainstate.m_from_snapshot_blockhash);
+
+    const std::optional<fs::path> chaindir = snapshot_chainstate.CoinsDB().StoragePath();
+    assert(chaindir); // Sanity check that chainstate isn't in-memory.
+    const fs::path write_to = *chaindir / node::SNAPSHOT_BLOCKHASH_FILENAME;
+
+    FILE* file{fsbridge::fopen(write_to, "wb")};
+    CAutoFile afile{file, SER_DISK, CLIENT_VERSION};
```

Comment:
nit in d59db4ff4b585d2b86b68ea0db206ec1c0cc5022:

```diff
diff --git a/src/node/utxo_snapshot.cpp b/src/node/utxo_snapshot.cpp
index 9c9947df95..2c5885198b 100644
--- a/src/node/utxo_snapshot.cpp
+++ b/src/node/utxo_snapshot.cpp
@@ -25,7 +25,7 @@ bool WriteSnapshotBaseBlockhash(CChainState& snapshot_chainstate)
     const fs::path write_to = *chaindir / node::SNAPSHOT_BLOCKHASH_FILENAME;
 
     FILE* file{fsbridge::fopen(write_to, "wb")};
-    CAutoFile afile{file, SER_DISK, CLIENT_VERSION};
+    AutoFile afile{file};
     if (afile.IsNull()) {
         LogPrintf("[snapshot] failed to open base blockhash file for writing: %s\n",
                   fs::PathToString(write_to));
@@ -60,7 +60,7 @@ std::optional<uint256> ReadSnapshotBaseBlockhash(fs::path chaindir)
 
     uint256 base_blockhash;
     FILE* file{fsbridge::fopen(read_from, "rb")};
-    CAutoFile afile{file, SER_DISK, CLIENT_VERSION};
+    AutoFile afile{file};
     if (afile.IsNull()) {
         LogPrintf("[snapshot] failed to open base blockhash file for reading: %s\n",
             read_from_str);

Reply:
I think it could be valuable to document `CAutoFile` in `streams.h`, there is no document to explain the diff compared to `AutoFile`. From my understanding, it's a wrapper to handle file pointer, without having to bother with serialization type and versioning.